File: ts/512/392.ts
function isSubsequence(s: string, t: string): boolean {
  const sl = s.length, tl = t.length
  let target = ''
  for (let i = 0, j = 0; i < sl; ++i) {
    while (j < tl) {
      if (s[i] === t[j]) {
        target += t[j++]
        break
      }
      j += 1
    }
    if (target.length < i + 1) return false
  }
  return true
};

File: py3/512/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        len_s, len_t = len(s), len(t)
        i, j = 0, 0
        while i < len_s and j < len_t:
            if s[i] == t[j]:
                i += 1
            j += 1

        return i == len_s